content/blog/accessing-dtap-in-pods.md

```diff
@@ -0,0 +1,211 @@
+pip install pyspark
+```
+There are two ways to interact with pyspark. The first one is to execute the ```pyspark``` command in bash to initiate the pyspark session. The second way is that to treat pyspark as a module which ```python``` kernel can import to. (```import pyspark```)
+
+### Method one: initiate ```pyspark``` session with jars
+In order to use datatap with pyspark, you have to add an external jar file as argument to pyspark. Initiate Spark's interactive shell in python using the following command. 
+```bash
+# bash
+
+# Specify the path of the jars files
+pyspark --jars /opt/bdfs/bluedata-dtap.jar
+```
+After starting the interactive shell, ```Spark Context``` and ```Spark Session``` are automatically initiate for you.
+![image](https://user-images.githubusercontent.com/72959956/120170783-e8d00680-c233-11eb-9fe8-136da9996fdc.png)
+
+We have to specify the Hadoop configurations. After that, you can read files from DataTap just like normally what you did with HDFS.
+```py
+# pyspark
+
+# Specify the Hadoop configurations.
+sc._jsc.hadoopConfiguration().set('fs.dtap.impl', 'com.bluedata.hadoop.bdfs.Bdfs')
+sc._jsc.hadoopConfiguration().set('fs.AbstractFileSystem.dtap.impl', 'com.bluedata.hadoop.bdfs.BdAbstractFS')
+
+# Commands for reading DataTap file.
+text = sc.textFile("dtap://TenantStorage/HPE.txt")
+text.take(5)
+```
+![image](https://user-images.githubusercontent.com/72959956/120171213-61cf5e00-c234-11eb-8928-2514e8b867a8.png)
+
+### Method two: initiate ```python``` and initiate ```pyspark``` with jars at runtime
+
+Run the Python Shell first:
+```bash
+# bash
+python3
+```
+
+At the Python runtime, add the path of the jar file using Spark configuration command:
+```py
+# python
+from pyspark import SparkConf, SparkContext
+
+# Specify the path of the jars files
+conf = SparkConf().set("spark.jars", "/opt/bdfs/bluedata-dtap.jar")
+sc = SparkContext( conf=conf)
+
+# Specify the Hadoop configurations.
+sc._jsc.hadoopConfiguration().set('fs.dtap.impl', 'com.bluedata.hadoop.bdfs.Bdfs')
+sc._jsc.hadoopConfiguration().set('fs.AbstractFileSystem.dtap.impl', 'com.bluedata.hadoop.bdfs.BdAbstractFS')
+
+# Commands for reading DataTap file.
+text = sc.textFile("dtap://TenantStorage/HPE.txt")
+text.take(5)
+```
+
+> References: 
+> [Spark Document: Runtime Environment](https://spark.apache.org/docs/latest/configuration.html#runtime-environment)
+> [Related GitHub Issues](https://github.com/delta-io/delta/issues/346)
+
+# Conclusion
+Distributed file system is fundamental for handling large amount of data. Managing those file systems are always pain for developer. DataTaps unify different storage resources into path that different clusters can use. This helps you to get rid of time-consuming copies or transfers of data. More time on extracting business insight from your data, less time on handling tedious stuff. That's what DataTap can provide you and that's what HPE Ezmeral can provide you.
```

Comment:
Change to "**A** **d**istributed file system is fundamental for handling large amount**s** of data."
Suggest changing "Managing those file systems are always pain for developer." to "Managing those file systems **tends to always be a pain** for developer**s**."

Change "...storage resources into path that different" to "storage resources into **a** path that different"

For the last sentences, recommend changing them to read: "More time spent on extracting business insight from your data and less time handling tedious stuff - that's what DataTap can give you. And that's what you get with HPE Ezmeral."

Reply:
Do you have any tags you want to include in this post?